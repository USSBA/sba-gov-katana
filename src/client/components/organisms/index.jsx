/* IMPORTS */

// miscellaenous organisms
import BusinessGuideTileCollection from "./business-guide-tile-collection/business-guide-tile-collection.jsx";
import CardCollection from "./card-collection/card-collection.jsx";
import DeveloperTester from "./developer-tester/developer-tester.jsx";
import DocumentCardCollection from "./document-card-collection/document-card-collection.jsx";
import FrontPageHero from "./front-page-hero/front-page-hero.jsx";
import FundingProgramsTileCollection from "./funding-programs-tile-collection/funding-programs-tile-collection.jsx";
import DisasterAlert from "./header-footer/disaster-alert/disaster-alert.jsx";
import NotificationBar from "./header-footer/notification-bar.jsx";
import Hero from "./hero/hero.jsx";
import RelatedDocumentCards from "./related-document-cards/related-document-cards.jsx";
import SearchBox from "./search-box/search-box.jsx";
import SectionNav from "./section-nav/section-nav.jsx";

// header-footer organisms
import Footer from "./header-footer/footer/footer.jsx";
import MobileNav from "./header-footer/mobile-nav/mobile-nav.jsx";
import Header from "./header-footer/header/header.jsx";

// homepage organisms
import Blog from "./homepage/blog/blog.jsx";
import FrontPageLady from "./homepage/front-page-lady/front-page-lady.jsx";
import HappeningNow from "./homepage/happening-now/happening-now.jsx";

// lender match organisms
// -- form
import AdditionalInfoForm from "./lender-match/form/additional-info.jsx";
import BusinessInfoForm from "./lender-match/form/business-info.jsx";
import ContactInfoForm from "./lender-match/form/contact-info.jsx";
import IndustryInfoForm from "./lender-match/form/industry-info.jsx";
import LoanInfo from "./lender-match/form/loan-info.jsx";
import ReviewSection from "./lender-match/form/review-page-helpers.jsx";
import ReviewSubmitInfoForm from "./lender-match/form/review-submit-info.jsx";
// -- landing page
import FindLendersIntro from "./lender-match/landing-page/find-lenders-intro.jsx";
import HelpfulQuestions from "./lender-match/landing-page/helpful-questions.jsx";
import {HowItWorksSection} from "./lender-match/landing-page/how-it-works.jsx";
import PreparationChecklist from "./lender-match/landing-page/preparation-checklist.jsx";
// -- success page
import ConfirmSection from "./lender-match/success-page/confirm-section.jsx";
import DynamicCounselingAndTools from "./lender-match/success-page/counseling-and-tools.jsx";
import CounselorMap from "./lender-match/success-page/counselor-map.jsx";
import EmailConfirmationInvalid from "./lender-match/success-page/email-confirmed-page.jsx";

// modal organisms
import LeaveSbaModal from "./modals/leave-sba-modal/leave-sba-modal.jsx";
import MobileSectionNavModal from "./modals/mobile-section-nav/mobile-section-nav.jsx";

/* EXPORTS */

// miscellaneous organisms
export {
	BusinessGuideTileCollection,
	CardCollection,
	DeveloperTester,
	DocumentCardCollection,
	FrontPageHero,
	FundingProgramsTileCollection,
	DisasterAlert,
	NotificationBar,
	Hero,
	RelatedDocumentCards,
	SearchBox,
	SectionNav
};

// header-footer organisms
export {
	Footer,
	MobileNav,
	Header
};

// homepage organisms
export {
	Blog,
	FrontPageLady,
	HappeningNow
};

// lender match organisms
export {
	AdditionalInfoForm,
	BusinessInfoForm,
	ContactInfoForm,
	IndustryInfoForm,
	LoanInfo,
	ReviewSection,
	ReviewSubmitInfoForm,
	FindLendersIntro,
	HelpfulQuestions,
	HowItWorksSection,
	PreparationChecklist,
	ConfirmSection,
	DynamicCounselingAndTools,
	CounselorMap,
	EmailConfirmationInvalid
};

// modal organisms
export {
	LeaveSbaModal,
	MobileSectionNavModal
};
