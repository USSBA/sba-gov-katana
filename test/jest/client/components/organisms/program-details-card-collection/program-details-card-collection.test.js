/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { ProgramDetailsCardCollection } from 'organisms'

function makeTestDataItem(itemNumber) {
  return {
    title: `Title ${itemNumber}`,
    url: `title-${itemNumber}`,
    fullUrl: `/section/subsction/title-${itemNumber}`,
    description: null,
    summary: `Summary ${itemNumber}`,
    node: `10${itemNumber}`,
    weight: -50,
    spanishTranslation: {
      title: `Título ${itemNumber}`,
      fullUrl: `titulo-${itemNumber}`,
      description: null,
      summary: `Resumen ${itemNumber}`
    }
  }
}

const testData = [makeTestDataItem(0), makeTestDataItem(1), makeTestDataItem(2), makeTestDataItem(3)]

describe('ProgramDetailsCardCollection', () => {
  test('should render the default cards', () => {
    const component = renderer.create(<ProgramDetailsCardCollection />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render nothing when given nothing', () => {
    const component = shallow(<ProgramDetailsCardCollection cards={null} />)
    expect(component.contains('CardCollection')).toEqual(false)
  })

  test('should render english by default', () => {
    const component = shallow(<ProgramDetailsCardCollection cards={testData} />)
    const cardsPropertyToCardCollection = component.find('CardCollection').prop('cards')
    // eslint-disable-next-line no-magic-numbers
    expect(cardsPropertyToCardCollection.length).toEqual(4)
    expect(cardsPropertyToCardCollection[0].titleText).toEqual('Title 0')
    expect(cardsPropertyToCardCollection[0].subtitleText).toEqual('Summary 0')
  })

  test('should render the spanish translation when the language is set', () => {
    const component = shallow(<ProgramDetailsCardCollection cards={testData} currentLanguage="es" />)
    const cardsPropertyToCardCollection = component.find('CardCollection').prop('cards')
    // eslint-disable-next-line no-magic-numbers
    expect(cardsPropertyToCardCollection.length).toEqual(4)
    expect(cardsPropertyToCardCollection[0].titleText).toEqual('Título 0')
    expect(cardsPropertyToCardCollection[0].subtitleText).toEqual('Resumen 0')
  })
})
