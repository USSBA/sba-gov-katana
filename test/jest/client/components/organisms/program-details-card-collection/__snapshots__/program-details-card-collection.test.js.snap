// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramDetailsCardCollection should render the default cards 1`] = `
<div>
  <div
    className="cardCollection"
  >
    <div
      className="cardRow"
      id="card-row--1-0"
    >
      <div
        aria-label="card 1"
        className="card sixColumn noImage"
        data-testid="card"
        id="card--1-1"
        tabIndex="0"
      >
        <div
          className="false"
        >
          <h4
            className="itemTitle"
            data-testid="card title"
            id="title--1-1"
          >
            <a
              onClick={[Function]}
              style={Object {}}
              target="_self"
            >
              Title A
            </a>
          </h4>
          <a
            data-testid="card link"
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Learn more
          </a>
        </div>
      </div>
      <div
        aria-label="card 2"
        className="card sixColumn noImage"
        data-testid="card"
        id="card--1-2"
        tabIndex="0"
      >
        <div
          className="false"
        >
          <h4
            className="itemTitle"
            data-testid="card title"
            id="title--1-2"
          >
            <a
              onClick={[Function]}
              style={Object {}}
              target="_self"
            >
              Title B
            </a>
          </h4>
          <a
            data-testid="card link"
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Learn more
          </a>
        </div>
      </div>
      <div
        aria-label="card 3"
        className="card sixColumn noImage"
        data-testid="card"
        id="card--1-3"
        tabIndex="0"
      >
        <div
          className="false"
        >
          <h4
            className="itemTitle"
            data-testid="card title"
            id="title--1-3"
          >
            <a
              onClick={[Function]}
              style={Object {}}
              target="_self"
            >
              Title C
            </a>
          </h4>
          <a
            data-testid="card link"
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Learn more
          </a>
        </div>
      </div>
      <div
        aria-label="card 4"
        className="card sixColumn noImage"
        data-testid="card"
        id="card--1-4"
        tabIndex="0"
      >
        <div
          className="false"
        >
          <h4
            className="itemTitle"
            data-testid="card title"
            id="title--1-4"
          >
            <a
              onClick={[Function]}
              style={Object {}}
              target="_self"
            >
              Title D
            </a>
          </h4>
          <a
            data-testid="card link"
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Learn more
          </a>
        </div>
      </div>
      <div
        aria-label="card 5"
        className="card sixColumn noImage"
        data-testid="card"
        id="card--1-5"
        tabIndex="0"
      >
        <div
          className="false"
        >
          <h4
            className="itemTitle"
            data-testid="card title"
            id="title--1-5"
          >
            <a
              onClick={[Function]}
              style={Object {}}
              target="_self"
            >
              Title E
            </a>
          </h4>
          <a
            data-testid="card link"
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Learn more
          </a>
        </div>
      </div>
      <div
        aria-label="card 6"
        className="card sixColumn noImage"
        data-testid="card"
        id="card--1-6"
        tabIndex="0"
      >
        <div
          className="false"
        >
          <h4
            className="itemTitle"
            data-testid="card title"
            id="title--1-6"
          >
            <a
              onClick={[Function]}
              style={Object {}}
              target="_self"
            >
              Title F
            </a>
          </h4>
          <a
            data-testid="card link"
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Learn more
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
